app/models/post.rb

```diff
@@ -1,7 +1,7 @@
 class Post < ActiveRecord::Base
   belongs_to :user
-  has_many :favorite_handlers
-  has_many :user_favorites, through: :favorite_handlers, source: "user"
+  has_many :favorites
+  has_many :user_favorites, through: :favorites, source: "user"
```

Comment:
This name makes it seem like this returns Favorite objects, but I believe it returns User objects. It should end with `_users`, or at least have `users` in somewhere. Off the top of my head, maybe `:favoriting_users` `:favorited_by_users` `:users_that_favorited`. Those all sound weird but are a little more clear.